src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt

```diff
@@ -73,4 +75,25 @@ object ValidationUtil {
         if(name.startsWith("."))
             validationException.addValidationError("Value $name must not start with '.'")
     }
-}
\ No newline at end of file
+
+    /**
+     * validate leader index version for compatibility
+     * If on higher version - Replication will not be allowed
+     *  - Upgrade path - Upgrade Follower cluster to higher version
+     *    and then upgrade leader
+     */
+    fun validateLeaderIndexMetadata(leaderIndexMetadata: IndexMetadata) {
+        if(Version.CURRENT.before(leaderIndexMetadata.creationVersion)) {
+            val err = "Leader index[${leaderIndexMetadata.index.name}] is on " +
+                    "higher version [${leaderIndexMetadata.creationVersion}] than follower [${Version.CURRENT}]"
+            log.error(err)
+            throw IllegalArgumentException(err)
+        }
+        if(Version.CURRENT.before(leaderIndexMetadata.upgradedVersion)) {
+            val err = "Leader index[${leaderIndexMetadata.index.name}] is upgraded with " +
+                    "higher version [${leaderIndexMetadata.creationVersion}] than follower [${Version.CURRENT}]"
```

Comment:
log should probably be leaderIndexMetadata.upgradedVersion

Reply:
Done